components/Categories.php

```diff
@@ -103,8 +103,15 @@ protected function loadCategories()
 
     protected function linkCategories($categories)
     {
-        return $categories->each(function($category) {
-            $category->setUrl($this->categoryPage, $this->controller);
+        $blogPostsComponent = $this->getComponent('blogPosts', $this->categoryPage);
+
+        return $categories->each(function($category) use ($blogPostsComponent) {
+            $category->setUrl(
+                $this->categoryPage,
+                $this->controller,
+                [
+                    'slug' => $this->urlProperty($blogPostsComponent, 'categoryFilter')
+                ]);
```

Comment:
@GinoPane slight code tweak, please make sure closing brackets for multi-line arguments are the same indentation as the original method call, as per PSR-2 guidelines.
```suggestion
                ]
            );
```